File: src/_0_intro/_02_smooth_sailing/AllLongestStrings.java
/*
 * Authored by: Jason Wesley Howse
 */

package _0_intro._02_smooth_sailing;

import java.util.ArrayList;

public class AllLongestStrings {
/*
Given an array of strings, return another array
containing all of its longest strings.
 */

    String[] allLongestStrings(String[] inputArray) {
        ArrayList<String> output = new ArrayList<>();
        output.add(inputArray[0]);
        for(int i=1;i<inputArray.length;i++){
            if(output.get(0).length()<inputArray[i].length()){
                output=new ArrayList<>();
                output.add(inputArray[i]);
            }else if(output.get(0).length()==inputArray[i].length()){//if(output.get(0).length()<inputArray[i].length()){
                output.add(inputArray[i]);
            }//else if(output.get(0).length()==inputArray[i].length()){
        }//for(int i=1;i<inputArray.length;i++){
        return output.toArray(new String[0]);
    }//String[] allLongestStrings(String[] inputArray) {

/*
Input/Output

[execution time limit] 3 seconds (java)

[input] array.string inputArray

A non-empty array.

Guaranteed constraints:
1 ≤ inputArray.length ≤ 10,
1 ≤ inputArray[i].length ≤ 10.

[output] array.string

Array of the longest strings, stored in the same order as in the inputArray.
 */

}//public class AllLongestStrings {

File: src/_0_intro/_02_smooth_sailing/AllLongestStrings.cs
using System.Collections.Generic;

namespace Algorithms.src._0_intro._02_smooth_sailing {
    public class AllLongestStrings {

        /*

Given an array of strings, return another array containing all of its longest strings.

Example

For inputArray = ["aba", "aa", "ad", "vcd", "aba"], the output should be
allLongestStrings(inputArray) = ["aba", "vcd", "aba"].

Input/Output

[execution time limit] 3 seconds (cs)

[input] array.string inputArray

A non-empty array.

Guaranteed constraints:
1 ≤ inputArray.length ≤ 10,
1 ≤ inputArray[i].length ≤ 10.

[output] array.string

Array of the longest strings, stored in the same order as in the inputArray.

        */

        public
        string[] allLongestStrings(string[] inputArray) {
            List<string> output = new List<string>();
            int maxLength = 0;
            foreach (var a in inputArray) {
                if (maxLength < a.Length) {
                    maxLength = a.Length;
                    output.Clear();
                }//if (maxLength < a.Length) {
                if (maxLength == a.Length) {
                    output.Add(a);
                }//if (maxLength == a.Length) {
            }//foreach(var a in inputArray) {
            return output.ToArray();
        }//string[] allLongestStrings(string[] inputArray) {
    }//public class AllLongestStrings {
}//namespace Algorithms.src._0_intro._02_smooth_sailing {